examples/ebitengine/main.go

```diff
@@ -0,0 +1,116 @@
+package main
+
+import (
+	"bytes"
+	"log/slog"
+	"runtime/debug"
+	"unsafe"
+
+	"github.com/hajimehoshi/ebiten/v2"
+	"github.com/hajimehoshi/ebiten/v2/text/v2"
+	"github.com/totallygamerjet/clay"
+	"github.com/totallygamerjet/clay/examples/fonts"
+	sl "github.com/totallygamerjet/clay/examples/shared-layouts"
+	"github.com/totallygamerjet/clay/renderers/ebitengine"
+)
+
+func handleClayError(errorText clay.ErrorData) {
+	slog.Error(errorText.ErrorText.String(), "stacktrace", debug.Stack())
+}
+
+// TODO: CreateArenaWithCapacityAndMemory should take a slice of bytes
+
+const (
+	winWidth, winHeight = 640, 480
+)
+
+type App struct {
+	demoData     sl.ClayVideoDemo_Data
+	cmds         clay.RenderCommandArray
+	rendererData *ebitengine.RendererData
+
+	width  int
+	height int
+}
+
+func New(demoData sl.ClayVideoDemo_Data, rendererData *ebitengine.RendererData) *App {
+	return &App{
+		demoData:     demoData,
+		rendererData: rendererData,
+	}
+}
+
+func (a *App) Update() error {
+	dx, dy := ebiten.Wheel()
+	if dx != 0 || dy != 0 {
+		clay.UpdateScrollContainers(true, clay.Vector2{
+			X: float32(dx),
+			Y: float32(dy),
+		}, 0.01)
+	}
+
+	x, y := ebiten.CursorPosition()
+	clay.SetPointerState(clay.Vector2{
+		X: float32(x),
+		Y: float32(y),
+	}, ebiten.IsMouseButtonPressed(ebiten.MouseButtonLeft))
+
+	a.cmds = sl.ClayVideoDemo_CreateLayout(&a.demoData)
+
+	return nil
+}
+
+func (a *App) Draw(screen *ebiten.Image) {
+	a.rendererData.Screen = screen
+	ebitengine.ClayRender(a.rendererData, a.cmds)
+}
+
+func (a *App) Layout(outsideWidth, outsideHeight int) (int, int) {
+	if outsideWidth != a.width || outsideHeight != a.height {
+		clay.SetLayoutDimensions(clay.Dimensions{
+			Width:  float32(outsideWidth),
+			Height: float32(outsideHeight),
+		})
+	}
+	a.width = outsideWidth
```

Comment:
I think the ugly text might be related to display scaling?

https://github.com/tinne26/etxt/blob/main/docs/display-scaling.md

Reply:
Fixed! (it was a color scaling issue, but the device scale factor also has to be taken into account for some machines)

Reply:
The scale is working but it's not being accounted for in the MeasureText and when rendering so the text is tiny on my Mac 
![image](https://github.com/user-attachments/assets/214faf6a-d057-457f-83bd-bfe1d01f6302)

Reply:
Maybe it's easier to see in this image. Perhaps the Ebitengine one is right and just needs a bigger font?
![image](https://github.com/user-attachments/assets/41a91643-22bf-4e6f-ab5b-03823e1bae0e)

Reply:
The font looks good but now everything else is crushed?

![image](https://github.com/user-attachments/assets/7d21c7f3-c883-454d-ace4-554cac4eb72c)